geopyspark/geotrellis/__init__.py

```diff
@@ -227,23 +235,32 @@ class TemporalProjectedExtent(namedtuple("TemporalProjectedExtent", 'extent inst
 
     Args:
         extent (:class:`~geopyspark.geotrellis.Extent`): The area the raster represents.
-        instance (int): The time stamp of the raster.
+        instant (``datetime.datetime``): The time stamp of the raster. This is
+            an instance of ``datetime.datetime`` that references the correct date and time.
         epsg (int, optional): The EPSG code of the CRS.
         proj4 (str, optional): The Proj.4 string representation of the CRS.
 
     Attributes:
         extent (:class:`~geopyspark.geotrellis.Extent`): The area the raster represents.
-        instance (int): The time stamp of the raster.
+        instant (int): The time stamp of the raster. This is an ``int`` which represents the
+            timestamp in miliseconds.
         epsg (int, optional): The EPSG code of the CRS.
         proj4 (str, optional): The Proj.4 string representation of the CRS.
 
     Note:
+        The ``instant`` attribute is in milliseconds since the 1970 epoch. Thus, it may be
+        different than the timestamp given by the ``datetime.datetime`` instance.
+
+    Note:
         Either ``epsg`` or ``proj4`` must be defined.
     """
 
     __slots__ = []
 
     def __new__(cls, extent, instant, epsg=None, proj4=None):
+        if isinstance(instant, datetime.datetime):
+            instant = _convert_to_unix_time(instant)
```

Comment:
Should this conversion happen in ProtoBuf decoders ? That way the python meaningful `datetime` instance can hang around as long as possible.